packages/access-api/test/helpers/ucanto-test-utils.js

```diff
@@ -0,0 +1,116 @@
+import * as Ucanto from '@ucanto/interface'
+import { Voucher } from '@web3-storage/capabilities'
+import * as assert from 'assert'
+
+/**
+ * Tests using context from "./helpers/context.js", which sets up a testable access-api inside miniflare.
+ *
+ * @param {() => Promise<{ issuer: Ucanto.Signer<Ucanto.DID<'key'>>, service: Ucanto.Signer<Ucanto.DID>, conn: Ucanto.ConnectionView<Record<string, any>> }>} createContext
+ * @param {object} [options]
+ * @param {Iterable<Promise<Ucanto.Principal>>} options.registerSpaces - spaces to register in access-api. Some access-api functionality on a space requires it to be registered.
+ */
+export function createTesterFromContext(createContext, options) {
```

Comment:
one of the reasons it's so gnarly is to afford for cases where a promise of a ucanto signer is created, and then that's used to build several testers, and all in a way that I can avoid any `await` before calling mocha `it` to have many separate `it` calls like you like (and I like too now).

I'm down to refactor things in there, but I may need a pair programming session with you to do it, so that's why I'm not doing now.